guava/src/com/google/common/base/Ascii.java

```diff
@@ -629,4 +630,155 @@ private static int getAlphaIndex(char c) {
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @param string    the string to be searched in.
+   * @param subString the substring to search for.
+   * @param fromIndex the index from which to start the search.
+   * @return the index of the first occurrence of the {@code subString}, or {@code -1} if there is
+   *     no such occurrence.
+   *
+   * @since NEXT
+   */
+  public static int indexOfIgnoreCase(String string, String subString, int fromIndex) {
+    return indexOfIgnoreCase(string.toCharArray(), 0, string.length(),
+        subString.toCharArray(), 0, subString.length(), fromIndex);
+  }
+
+  /**
+   * Returns the index within the {@code string} of the first occurrence of the {@code subString},
+   * ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @param string the string to be searched in.
+   * @param subString the substring to search for.
+   * @return the index of the first occurrence of the {@code subString},
+   *     or {@code -1} if there is no such occurrence.
+   *
+   * @since NEXT
+   */
+  public static int indexOfIgnoreCase(String string, String subString) {
+    return indexOfIgnoreCase(string, subString, 0);
+  }
+
+  /**
+   * Indicates whether the character sequence {@code sequence} contains the {@code subSequence},
+   * ignoring the case of any ASCII alphabetic characters between {@code 'a'} and {@code 'z'}
+   * or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @since NEXT
+   */
+  public static boolean containsIgnoreCase(CharSequence sequence, CharSequence subSequence) {
+    // Calling length() is the null pointer check (so do it before we can exit early).
+    int length = sequence.length();
+    if (sequence == subSequence) {
+      return true;
+    }
+    // if subSequence is longer than sequence, it is impossible for sequence to contain subSequence
+    if (subSequence.length() > length) {
+      return false;
+    }
+    return indexOfIgnoreCase(sequence.toString(), subSequence.toString()) > -1;
+  }
+
+  /**
+   * Returns if the character sequence {@code seq} starts with the character sequence {@code prefix}
+   * at the specified {@code offset}, ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @since NEXT
+   */
+  public static boolean startsWithIgnoreCase(CharSequence seq, CharSequence prefix, int offset) {
+    // an empty prefix starts everywhere in the sequence
+    if (prefix.length() == 0) {
+      return true;
+    }
+    // if the prefix is longer than the sequence itself or from the offset, it is impossible
+    // for the sequence to contain the prefix
+    if (prefix.length() > seq.length() || prefix.length() > seq.length() - offset) {
+      return false;
+    }
+    return indexOfIgnoreCase(seq.toString(), prefix.toString()) == offset;
```

Comment:
This one is
1. Logically flawed
2. Too slow (A single for loop can fix it)

Reply:
Sounded like a good idea at first but now I agree - you guys are 100% correct. Will fix that.